packages/experimental/Switch/src/Switch.tsx

```diff
@@ -29,19 +31,108 @@ export const switchLookup = (layer: string, state: SwitchState, userProps: Switc
   );
 };
 
+// Function to change background slowly over time when switch is toggled.
+const startTrackBackgroundAnimation = (checked: boolean, animation: Animated.Value) => {
+  const toValue = checked ? 0 : 1;
+  Animated.timing(animation, {
+    toValue,
+    duration: 500,
+    useNativeDriver: false,
+  }).start();
+};
+
+// Function to transform the knob over track in animated way when switch is toggled.
+const startTrackAnimation = (onInit = false, animationConfig: IAnimationConfig, animation: Animated.Value, toggled: boolean) => {
+  Animated.timing(animation, {
+    toValue: toggled
+      ? animationConfig.trackWidth - (animationConfig.thumbWidth + animationConfig.thumbMargin * 2)
+      : onInit
+      ? 0
+      : -(animationConfig.trackWidth + animationConfig.thumbWidth),
+    duration: 300,
+    useNativeDriver: false,
+  }).start();
+};
+
+const isMobile = Platform.OS === 'android' || Platform.OS == 'ios';
+
 export const Switch = compose<SwitchType>({
   displayName: switchName,
   ...stylingSettings,
   slots: {
     root: Pressable,
     label: Text,
-    track: View,
-    thumb: View,
+    track: Animated.View, // Conversion from View to Animated.View for Animated API to work
+    thumb: Animated.View,
     toggleContainer: View,
     onOffText: Text,
   },
   useRender: (userProps: SwitchProps, useSlots: UseSlots<SwitchType>) => {
-    const switchInfo = useSwitch(userProps);
+    const switchOnSlot = useSlots(userProps, (layer) => switchLookup(layer, { toggled: true, disabled: userProps.disabled }, {}));
+    const switchOffSlot = useSlots(userProps, (layer) => switchLookup(layer, { toggled: false, disabled: userProps.disabled }, {}));
+
+    const animationConfig: IAnimationConfig = {
+      toggleOnBgColor: switchOnSlot.track({}).props.style.backgroundColor,
+      toggleOffBgColor: switchOffSlot.track({}).props.style.backgroundColor,
+      trackWidth: switchOnSlot.track({}).props.style.width,
+      thumbWidth: switchOnSlot.thumb({}).props.style.width,
+      thumbMargin: switchOnSlot.thumb({}).props.style.margin,
+    };
+
+    const [animation] = React.useState(new Animated.Value(0));
+    const [trackBackgroundAnimation] = React.useState(new Animated.Value(0));
+
+    const onChangeWithAnimation = React.useCallback(
+      (e: InteractionEvent, checked?: boolean) => {
+        userProps.onChange && userProps.onChange(e, checked);
+        if (isMobile) {
+          if (checked) {
+            startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
+            startTrackAnimation(false, animationConfig, animation, checked);
+          } else {
+            startTrackBackgroundAnimation(checked, trackBackgroundAnimation);
+            startTrackAnimation(false, animationConfig, animation, checked);
+          }
+        }
+      },
+      [userProps.onChange],
```

Comment:
should include trackBackgroundAnimation, animation

Reply:
added